src/auth/awscredentialsclient.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GaxiosOptions} from 'gaxios';
+
+import {AwsClientOptions} from './awsclient';
+import {AwsRequestSigner} from './awsrequestsigner';
+import {
+  BaseExternalAccountClient,
+} from './baseexternalclient';
+import {RefreshOptions} from './oauth2client';
+
+/**
+ * AWS SDK v3 compatible type.
+ */
+export interface AwsCredentials {
+  /**
+   * AWS access key ID
+   */
+  readonly accessKeyId: string;
+  /**
+   * AWS secret access key
+   */
+  readonly secretAccessKey: string;
+  /**
+   * A security or session token to use with these credentials. Usually
+   * present for temporary credentials.
+   */
+  readonly sessionToken?: string;
+  /**
+   * A {Date} when these credentials will no longer be accepted.
+   */
+  readonly expiration?: Date;
+}
```

Comment:
Copied from https://github.com/aws/aws-sdk-js-v3/blob/main/packages/types/src/credentials.ts#L3-L27.
We can also use `@aws-sdk/types` [![](https://img.shields.io/badge/npm-@aws--sdk%2Ftypes-%23cb0000?style=flat&logo=npm)](https://www.npmjs.com/package/@aws-sdk/types).
Is it needed to omit comment documents for copyright reasons?